README.md

```diff
@@ -805,6 +805,38 @@ struct FetchMoviesPhaseAtom: ValueAtom, Refreshable, Hashable {
 }
 ```
 
+
+#### [Resettable](https://ra1028.github.io/swiftui-atom-properties/documentation/atoms/resettable)
+
+`Resettable` allows you to implement additional reset behavior to an atom.
+
+<details><summary><code>📖 Expand to see example</code></summary>
+
+It adds custom reset behavior to an Atom that will be executed prior to new cache generation.
+
+It's useful when need to handle reset side-effects or when utilizing private atom from other atoms.
+  
+In following example, `RandomIntAtom` generates a random value using private `RandomNumberGeneratorAtom`, and `Resettable` gives ability to reset `RandomNumberGeneratorAtom` with new randomizer seed.
+
+```swift
+struct RandomIntAtom: ValueAtom, Resettable, Hashable {
+    func value(context: Context) -> Int {
+        let generator = context.watch(RandomNumberGeneratorAtom())
+        return .random(in: 0..<100, using: $generator)
+    }
+
+    func reset(context: ResetContext) {
+        context.reset(RandomNumberGeneratorAtom())
```

Comment:
This example looks wrong to me. RandomNumberGeneratorAtom caches only `SystemRandomNumberGenerator` instance and the value of `RandomIntAtom` isn't updated even if you reset it.

Reply:
Do you mean as a concept? 

Because of `context.watch(RandomNumberGeneratorAtom())` in the value, Resetting `RandomNumberGeneratorAtom` would definitely call notify `RandomIntAtom` to update itself.

```swift
func value(context: Context) -> Int {
        let generator = context.watch(RandomNumberGeneratorAtom())
        return .random(in: 0..<100, using: $generator)
    }
```


You can test this with:


```swift
func testCustomReset() {
        let context = AtomTestContext()

        let resettableAtom = TestCustomResettableAtom(
            defaultValue: { context in
                var generator = context.watch(RandomNumberGeneratorAtom())
                return Int.random(in: 0..<100, using: &generator)
            },
            reset: { context in
                // comment below line to see difference
                context.reset(RandomNumberGeneratorAtom())
            }
        )

        let initialValue = context.watch(resettableAtom)
        print("initialValue: \(initialValue)")

        context.reset(resettableAtom)

        let postResetValue = context.watch(resettableAtom)
        print("postResetValue: \(postResetValue)")
        
        XCTAssertNotEqual(initialValue, postResetValue)
    }

struct TestCustomResettableAtom<T>: StateAtom, Resettable {
    var defaultValue: (Context) -> T
    var reset: (ResetContext) -> Void

    var key: UniqueKey {
        UniqueKey()
    }

    func defaultValue(context: Context) -> T {
        defaultValue(context)
    }

    func reset(context: ResetContext) {
        reset(context)
    }
}

private struct RandomNumberGeneratorAtom: ValueAtom, Hashable {
    func value(context: Context) -> SystemRandomNumberGenerator {
        return SystemRandomNumberGenerator()
    }
}

```

Reply:
Oh, yes. That's true I guess I missed something. Sorry for confusion.